iothub/device/src/ClientFactory.cs

```diff
@@ -385,6 +379,17 @@ internal static ITransportSettings[] GetTransportSettings(TransportType transpor
                 }
             }
 
+            if (authenticationMethod is DeviceAuthenticationWithX509Certificate)
+            {
+                if (builder.Certificate == null)
+                {
+                    throw new ArgumentException("certificate must be present in DeviceAuthenticationWithX509Certificate");
```

Comment:
nit: can we make the exception message descriptive, instead of referencing the `DeviceAuthenticationWithX509Certificate` class type :)